From 7ffcf5479eb74c4b55e5dd3d7f10940225855182 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Fri, 20 Mar 2015 03:29:28 +0800
Subject: [PATCH] MLK-10439: arm: imx: fix system hang after resume back under
 low busfreq

System will be no response after resume back under low busfreq mode. The
root cause of this issue is that when the system is under ultra_low_bus_freq
mode on i.MX6SL, resume process has a low bus mode deamon in background,
cause system enter low bus mode twice, busfreq driver will notify thermal
driver to turn off PLL3 twice, and cause PLL3 usecount incorrect and UART
no message out.

Signed-off-by: Bai Ping <b51503@freescale.com>
(cherry picked from commit 91cf351a2afc17ac4a260e4d2ad1e32d00925a1b)
---
 arch/arm/mach-imx/busfreq-imx6.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/mach-imx/busfreq-imx6.c b/arch/arm/mach-imx/busfreq-imx6.c
index 7d65d08..af7982f 100644
--- a/arch/arm/mach-imx/busfreq-imx6.c
+++ b/arch/arm/mach-imx/busfreq-imx6.c
@@ -786,7 +786,7 @@ void __init imx6_busfreq_map_io(void)
 static void bus_freq_daemon_handler(struct work_struct *work)
 {
 	mutex_lock(&bus_freq_mutex);
-	if ((!low_bus_freq_mode) && (high_bus_count == 0) &&
+	if ((!low_bus_freq_mode) && (!ultra_low_bus_freq_mode) && (high_bus_count == 0) &&
 		(med_bus_count == 0) && (audio_bus_count == 0))
 		set_low_bus_freq();
 	mutex_unlock(&bus_freq_mutex);
-- 
1.8.0

